base/io.jl

```diff
@@ -616,8 +616,7 @@ Logan
 """
 readline(filename::AbstractString; keep::Bool=false) =
     open(io -> readline(io; keep), filename)
-readline(s::IO=stdin; keep::Bool=false) =
-    String(_unsafe_take!(copyline(IOBuffer(sizehint=16), s; keep)))
+readline(s::IO=stdin; keep::Bool=false) = takestring!(copyline(IOBuffer(sizehint=16), s; keep))
```

Comment:
```suggestion
readline(s::IO=stdin; keep::Bool=false) = unsafe_takestring!(copyline(IOBuffer(sizehint=16), s; keep))
```

Reply:
There is some discussion to be had, since `IO` is generic here, an implementation of `copyline` could store a reference to the `IOBuffer` somewhere and use it later. However, that does seem unlikely.

Reply:
Ah, okay.  We were using `_unsafe_take!` before so I wasn't sure why it was removed.